# This file is part of HiGFXback

# requires
REQUIRES="make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --disable-nls --prefix=

# build
make

# install
install -d $DESTDIR/bin
install dos2unix $DESTDIR/bin
install unix2dos $DESTDIR/bin

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/dos2unix-build.pc << EOF
Name: dos2unix
Version: 7.4.4
Description: DOS to Unix and vice versa text file format converter
Requires: $REQUIRES

devel=\\
/bin/dos2unix \\
/bin/unix2dos
EOF

exit

# patch
--- dos2unix-7.4.4.orig/Configure
+++ dos2unix-7.4.4/Configure
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+prefix=/usr/local
+
+ENABLE_NLS=1
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) prefix=${1#*=};;
+    --disable-nls) ENABLE_NLS=;;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+sed -i "/^prefix/s|=.*|= $prefix|" Makefile
+
+sed -i "/^ENABLE_NLS/s|=.*|= $ENABLE_NLS|" Makefile
--- dos2unix-7.4.4.orig/Makefile
+++ dos2unix-7.4.4/Makefile
@@ -70,7 +70,7 @@
 LINK            = ln -sf
 LINK_MAN        = $(LINK)
 
-prefix          = /usr
+prefix          = /usr/local
 exec_prefix     = $(prefix)
 bindir          = $(exec_prefix)/bin
 datarootdir     = $(prefix)/share
--- dos2unix-7.4.4.orig/common.c
+++ dos2unix-7.4.4/common.c
@@ -778,6 +778,8 @@
 #endif
 #ifdef ENABLE_NLS
   D2U_ANSI_FPRINTF(stdout,"LOCALEDIR: %s\n", localedir);
+#else
+  (void) localedir;
 #endif
   D2U_ANSI_FPRINTF(stdout,"https://waterlan.home.xs4all.nl/dos2unix.html\n");
   D2U_ANSI_FPRINTF(stdout,"https://dos2unix.sourceforge.io/\n");

# source
https://waterlan.home.xs4all.nl/dos2unix/dos2unix-7.4.4.tar.gz
